src/chat_manager.ts

```diff
@@ -68,11 +69,34 @@ export default class ChatManager {
   }
 
   connect(options: ConnectOptions) {
+    const cursorsReq: Promise<any> = this.userId
+      ? this.cursorsInstance
+          .request({
+            method: 'GET',
+            path: `/cursors/0/users/${this.userId}`,
+          })
+          .then(res => {
+            const cursors = JSON.parse(res);
+            const cursorsByRoom: { [roomId: number]: BasicCursor } = {};
+            cursors.forEach((c: any): void => {
+              cursorsByRoom[
+                c.room_id
+              ] = PayloadDeserializer.createBasicCursorFromPayload(c);
+            });
+            return cursorsByRoom;
+          })
+      : Promise.resolve({});
+
     this.userSubscription = new UserSubscription({
       apiInstance: this.apiInstance,
       connectCompletionHandler: (currentUser?: CurrentUser, error?: any) => {
         if (currentUser) {
-          options.onSuccess(currentUser);
+          cursorsReq
+            .then(cursors => {
```

Comment:
Waiting on the `cursorsReq` promise here feels a bit strange in that the rest of the SDK might have already continued on with stuff like setting up the presence subscription etc, but the actual "connection" gets held up here waiting for the cursors. Do we think it's necessary to wait on the cursors being available here as opposed to doing something that means room subscriptions get (perhaps optionally) held up by the cursors request?

Reply:
Hmmmm holding up room subscriptions might be nicer yeah. In practice I think it makes very little difference. I ran some timers while I was testing this and the cursors request always resolves before the user subscription opens, but I'll see what it looks like to pass the promise forward and hold up the room subscriptions instead. (or not, in practice, since the promise will have resolved long ago 😛)